cider/include/cider/batch/CiderBatch.h

```diff
@@ -108,6 +108,10 @@ class CiderBatch {
 
   const void** getChildrenArrayPtr() const;
 
+  void setArrowSchema(const ArrowSchema* arrow_schema) { arrow_schema_ = const_cast<ArrowSchema*>(arrow_schema); }
+
+  void setArrowArray(const ArrowArray* arrow_array) {arrow_array_ = const_cast<ArrowArray*>(arrow_array); }
+
```

Comment:
Need to add setter api instead of assigning member variables with a constructor?